docs/source/modin/performance.rst

```diff
@@ -0,0 +1,50 @@
+Performance Recommendations
+===========================
+
+This page contains recommendations to help improve performance when using Snowpark pandas.
+
+Caching Intermediate Results
+----------------------------
+Snowpark pandas uses a lazy paradigm - when operations are called on a Snowpark pandas object,
```

Comment:
That makes sense - my one concern is that this may be too detailed for users? Although I suppose they're probably a power user if they're looking into performance optimizations, so its probably ok.

Reply:
I always liked as a user to get an explanation. If you think it's too detailed, we can always make it less prominent, or make it a note explaining it. Else, if you rather skip an explanation it's also ok to simply describe what the operation does. We can also compile a list of tips and tricks, e.g.:

```
1. Limit number of chained operations with caching
....
2. Select only columns you use
....
3. Avoid `to_pandas`, leave data server-side
....
4. Avoid df.iterrows(), use instead ...
...
```
That would look then more like a performance section!